packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -4251,6 +4259,12 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   @override
   InheritedWidget dependOnInheritedElement(InheritedElement ancestor, { Object? aspect }) {
     assert(ancestor != null);
+    assert(
+       !ancestor.clearDependencyOnRebuild || debugDoingBuild || !dirty,
```

Comment:
Can you explain the `!dirty`?

Reply:
It's meant to disable the assert outside of the widget life-cycles (so after `build` has completed)

That enables support for things like using `context. dependOnInheritedElement` inside `ListView.builder`/`LayoutBuilder` – as they have this "layout on depend" which happens after the widget's `build` method has completed

Reply:
Please add that explanation as a comment.

However, this will now also allow developers to call this in other handlers outside of build, e.g. a pointer event handler, no? It may actually be pretty confusing there. If it is called from an event handler while the widget is clean, it goes through without error. If it is called from the same event handler after the widget has already been marked as dirty otherwise, it will fail.